### PR TITLE
JERSEY-2583: Reusable user token in ApacheConnector preventing SSL hands...

### PR DESCRIPTION
**Pull request by cfiehe**
...hakes on every request when client authentiation is required and update to HttpClient 4.4.1.

### CODE REVIEW

**Comment by mgajdos** [comment]
Jenkins test this patch.

**Comment by mgajdos** [comment]
Hi @cfiehe, 
thanks for your contribution.

Before we can review your pull request, we need you to sign Oracle Contributor Agreement.

More information about contributing to Jersey: https://jersey.java.net/scm.html#/Submitting_Patches_and_Contribute_Code…

Regards,
Michal

**Comment by cfiehe** [comment]
The OCA has been submitted.

**Comment by AdamLindenthal** [comment]
Hi @cfiehe, 

do you already see yourself in the list? I would expect to find you there already, but I didn't see your nick listed.
If you do, under what name?

Thanks,
Adam

**Comment by cfiehe** [comment]
Are you referring to the list of contributors (http://www.oracle.com/technetwork/community/oca-486395.html#list)? If so, I cannot find myself in there, too. The nickname should be cfiehe. Should I resend the OCA?

**Comment by AdamLindenthal** [comment]
No, you should see yourself there by now :-) Now, please be patient a bit and wait until someone reviews your pull request. It's not easy to find a while these days, but we are trying! 

Regards,
Adam

**Comment by cfiehe** [comment]
Agreed! Thank you very much.

File: connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java [hunk 1]
```diff
@@ -669,4 +677,30 @@ protected OutputStream createOutputStream(final long len, final SessionOutputBuf
...
+                }
+            } else if (!Objects.equal(clientContext.getUserToken(), userToken)) {
+                userToken = clientContext.getUserToken();
+            }
+        }
+
+        @Override
+        public void process(HttpResponse response, HttpContext context) throws HttpException, IOException {
+            HttpClientContext clientContext = HttpClientContext.adapt(context);
+            if (!Objects.equal(userToken, clientContext.getUserToken())) {
```
Comment (shamoh) [inline connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java]: Can you use same arguments order as in line 692:

```
    if (!Objects.equal(clientContext.getUserToken(), userToken))
```

**Comment by shamoh** [comment]
@cfiehe Please, provide unit test.

**Pull request closed without merging**